crates/syntax/src/ast/edit_in_place.rs

```diff
@@ -257,6 +290,28 @@ impl ast::GenericParamList {
         }
     }
 
+    pub fn syntax_editor_add_generic_param(
+        &self,
+        editor: &mut SyntaxEditor,
+        new_param: ast::GenericParam,
+    ) {
+        match self.generic_params().last() {
+            Some(_) => {
+                let mut params =
+                    self.generic_params().map(|param| param.clone()).collect::<Vec<_>>();
+                params.push(new_param.into());
+                let new_param_list = make::generic_param_list(params);
+
+                editor.replace(self.syntax(), new_param_list.syntax());
```

Comment:
It also panics here with rowan's mutable syntax tree thing that the existing generic params list is not mutable.
As the current `SyntaxEditor` is transitional thing, we might have to mark it with something like `clone_for_update` or something I guess.
But I think that an in-place inserting like the following would also work
```suggestion
                let position = self.r_angle_token().map_or_else(
                    || crate::syntax_editor::Position::last_child_of(self.syntax()),
                    crate::syntax_editor::Position::before,
                );
                let elements = vec![
                    make::token(SyntaxKind::COMMA).into(),
                    make::token(SyntaxKind::WHITESPACE).into(),
                    new_param.syntax().clone().into(),
                ];
                editor.insert_all(position, elements);
```